app/parse.py

```diff
@@ -15,8 +29,55 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product: Tag) -> Product:
+    return Product(
+        title=product.select_one(".title")["title"],
+        description=product.select_one(".description").text.replace("\xa0", " "),
+        price=float(product.select_one(".price").text.replace("$", "")),
+        rating=len(product.select(".ratings span")),
+        num_of_reviews=int(product.select_one(".review-count").text.split()[0])
+    )
+
+
+def get_page_products(page_url: str, driver: WebDriver) -> list[Product]:
+    driver.get(page_url)
+    cookies = driver.find_elements(By.CLASS_NAME, "acceptCookies")
+
+    if cookies:
+        cookies[0].click()
+
+    button_more = driver.find_elements(By.CLASS_NAME, "ecomerce-items-scroll-more")
+
+    if button_more:
+        while button_more[0].is_displayed():
+            button_more[0].click()
+            time.sleep(0.1)
```

Comment:
https://selenium-python.readthedocs.io/waits.html#explicit-waits
try to use WebDriverWait